common/logisticspipes/gui/GuiGuideBook.java

```diff
@@ -0,0 +1,171 @@
+		/**This is the darker background like the inventory has. Check whether this is wanted or not.*/
+		GlStateManager.translate(this.width / 2, this.height / 2, 0.0F);
+		/**TopLeft*/
+		drawTexturedModalRect(0, 0, 0, 0, eight * 4, eight * 4);
+		xRem -= 4;
+		/**TopRight*/
+		drawTexturedModalRect(eight * (x - 4), 0, eight * 4, 0, eight * 4, eight * 4);
+		xRem -= 4;
+		/**BottomLeft*/
+		drawTexturedModalRect(0, eight * (y - 4), 0, eight * 4, eight * 4, eight * 4);
+		yRem -= 4;
+		/**BottomRight*/
+		yRem -= 4;
+		drawTexturedModalRect(eight * (x - 4), eight * (y - 4), eight * 4, eight * 4, eight * 4, eight * 4);
+		/** Top & Bottom/ */
+		for(int i = 0; i < xRem; i++){
+			drawTexturedModalRect(eight * (i + 4), 0, eight * 3, 0, eight * 2, eight * 4);
+			drawTexturedModalRect(eight * (i + 4), eight * (y - 4), eight * 3, eight * 4, eight * 2, eight * 4);
+		}
+		/**Left, Right & Background*/
+		for(int j = 0; j < yRem; j++) {
+			drawTexturedModalRect(0, eight * (j + 4), 0, eight * 3, eight * 4, eight * 2);
+			drawTexturedModalRect(eight * (x - 4), eight * (j + 4), eight * 4, eight * 3, eight * 4, eight * 2);
+			for(int i = 0; i < xRem; i++){
+				drawTexturedModalRect(eight * (i + 4), eight * (j + 4), eight * 3, eight * 3, eight * 2, eight * 2);
+			}
+		}
+		GlStateManager.popMatrix();
+	}
+
+	public void drawGuiForeGroundLayer(){
+		GlStateManager.pushMatrix();
+		GlStateManager.translate(1.0F, 1.0F, 10.0F);
+		drawCenteredString(this.fontRenderer, StringUtils.translate("book.quickstart.title").trim(), this.width / 2, (((-((this.height/4)*3)/8)*4)+this.height/2) + 4, 16777215);
+		drawCenteredString(this.fontRenderer, "Page " + PAGE + " out of " + PAGE_MAX, this.width / 2, this.height - 25, 16777215);
+		GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
+		String delimiter = "\\\\n";
+		String currentPage = StringUtils.translate("book.quickstart." + PAGE);
+		MAX_LENGTH = this.width - 75 - (((-((this.width/4)*3)/8)*4)+this.width/2)*2;
+		String[] currentPageLines = currentPage.split(delimiter);
+		String line;
+		int line_number = 0;
+		int length = 0;
+		for(String sentence : currentPageLines){
+			do{
+				line = fontRenderer.trimStringToWidth(sentence, MAX_LENGTH);
+				length = line.length();
+				if(fontRenderer.getStringWidth(sentence) > MAX_LENGTH){
+					line = fontRenderer.trimStringToWidth(sentence, MAX_LENGTH);
+					if(line.charAt(line.length()-1) != ' '){
+						if(line.charAt(line.length()-2) != ' '){
+							line = line.substring(0, line.length()-1) + '-';
+							length-=1;
+						}else{
+							line = line.substring(0, line.length()-1) + ' ';
+							length-=1;
+						}
+					}
+					if(line.charAt(0) == ' ') line = line.substring(1);
+					sentence = sentence.substring(length);
+				}
+				else{
+					line = sentence;
+					sentence = "";
+				}
+				drawString(this.fontRenderer, line, (((-((this.width/4)*3)/8)*4)+this.width/2) + 24, (((-((this.height/4)*3)/8)*4)+this.height/2) + 24 + (10 * line_number++), 16777215);
```

Comment:
same as above; `16777215` → `0xFFFFFF`

the width and height calculations look refactorable (`(((-((x/4)*3)/8)*4)+x/2) + 24` = `(int) (-23.5 * x) + 24`), but I don't know if they're equivalent or if this makes use of int division. Try it, though. Or at least remove some of those parens, Java has order of operations

Reply:
I know I have to spend some time simplifying these calculations I was just adding things as needed

Reply:
This equates to:
`x * 1/8 + 24`

Reply:
Oh whoops, yes that's correct.